cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java

```diff
@@ -337,15 +350,21 @@ private List<CloudVmInstanceStatus> fillCloudVmInstanceStatuses(AuthenticatedCon
         return cloudVmInstanceStatuses;
     }
 
-    private Collection<String> instanceIdsWhichAreNotInCorrectState(List<CloudInstance> vms, AmazonEc2Client amazonEC2Client, String state) {
+    private Collection<String> instanceIdsWhichAreNotInCorrectState(List<CloudInstance> vms,
+            AmazonEc2Client amazonEC2Client, String state) {
         Set<String> instances = vms.stream().map(CloudInstance::getInstanceId).collect(Collectors.toCollection(HashSet::new));
+        pendingInstances = vms.stream().map(CloudInstance::getInstanceId).collect(Collectors.toCollection(HashSet::new));
+        Set<String> status = PENDING_STATUSES.stream().map(i -> i.toString().toLowerCase()).collect(Collectors.toUnmodifiableSet());
```

Comment:
do we expect any other 'pending' like states? because we're doing a lot of hassle around putting a single enum into a set and then iterating over that one element set to obtain its lowercase value and putting it into another one element set

Reply:
Currently other than pending, I have seen "stopping" as another state where the nodes might be stuck and we will have to force stop them as well in that case. But this problem has not been encountered by any customer til now so I did not add that. But to make it future proof I made set so if any other such state comes up than we can directly add it there.